src/modules/mc_pos_control/Takeoff/Takeoff.cpp

```diff
@@ -111,22 +111,24 @@ void Takeoff::updateTakeoffState(const bool armed, const bool landed, const bool
 
 float Takeoff::updateRamp(const float dt, const float takeoff_desired_vz)
 {
+	float upwards_velocity_limit = takeoff_desired_vz;
+
 	if (_takeoff_state < TakeoffState::rampup) {
-		return _takeoff_ramp_vz_init;
+		upwards_velocity_limit = _takeoff_ramp_vz_init;
 	}
 
 	if (_takeoff_state == TakeoffState::rampup) {
 		if (_takeoff_ramp_time > dt) {
-			_takeoff_ramp_vz += (takeoff_desired_vz - _takeoff_ramp_vz_init) * dt / _takeoff_ramp_time;
+			_takeoff_ramp_progress += dt / _takeoff_ramp_time;
```

Comment:
I don't think it is a real concern, but in theory you can end up with `_takeoff_ramp_progress` > 1.0

Reply:
Numerically for one time iteration yes, you're right. I could simplify the implementation anyways now that the ramp is always linear. I think I'll just create a follow-up simplification pr for that.